lib/elixir/lib/kernel.ex

```diff
@@ -2676,10 +2676,10 @@ defmodule Kernel do
     end
 
     case right do
-      _ when in_module? ->
```

Comment:
For instance:
```elixir
db_insertion() in []
```
with new order will be rewritten as simple `false` and `db_insertion()` will never happen.